packages/liquidator/mocha-test/liquidationStrategy.js

```diff
@@ -0,0 +1,319 @@
+    });
+    // should respect our balance and whale defense fund
+    assert.equal(result.toString(), "5");
+
+    strat = Strategy(
+      {
+        ...config,
+        whaleDefenseFundWei: "95",
+        defenseActivationPercent: 50,
+        withdrawalLiveness: 1000,
+        minSponsorSize: 100
+      },
+      { toBN, BN }
+    );
+
+    result = strat.calculateTokensToLiquidate({
+      syntheticTokenBalance: 100,
+      positionTokens: 101,
+      maxTokensToLiquidateWei: "1000"
+    });
+    // should respsect empminsponsor size
+    assert.equal(result.toString(), "1");
+
+    result = strat.calculateTokensToLiquidate({
+      syntheticTokenBalance: 1000,
+      positionTokens: 10000,
+      maxTokensToLiquidateWei: "100"
+    });
+    // should respect max to liquidate
+    assert.equal(result.toString(), "100");
+  });
+  it("shouldLiquidate", () => {
+    const config = {
+      minSponsorSize: "10"
+    };
+    let strat = Strategy(config, { toBN, BN });
+    let result = strat.shouldLiquidate({ tokensToLiquidate: "100000" });
+    assert(result);
+    result = strat.shouldLiquidate({ tokensToLiquidate: "0" });
+    assert(!result);
+  });
+  it("shouldLiquidateMinimum", () => {
+    const config = {
+      defenseActivationPercent: 50,
+      minSponsorSize: "10",
+      liquidationLiveness: 1000
+    };
+    let strat = Strategy(config, { toBN, BN });
+    const position = {
+      withdrawalRequestPassTimestamp: 1000,
+      numTokens: "100"
+    };
+    let result = strat.shouldLiquidateMinimum({
+      position,
+      syntheticTokenBalance: "50",
+      currentBlockTime: 500
+    });
+    assert(result);
+    result = strat.shouldLiquidateMinimum({
+      position,
+      // our token balance is high enough to liquidate full position
+      // so this should not active minimum
+      syntheticTokenBalance: "100",
+      currentBlockTime: 500
+    });
+    assert(!result);
+    result = strat.shouldLiquidateMinimum({
+      position,
+      syntheticTokenBalance: "50",
+      // sponsor has not passed the withdraw liveness % complete to liquidate
+      currentBlockTime: 0
+    });
+    assert(!result);
+    result = strat.shouldLiquidateMinimum({
+      position,
+      syntheticTokenBalance: "50",
+      // withdraw has passed liveness
+      currentBlockTime: 1000
+    });
+    assert(!result);
+  });
+  it("processPosition", () => {
+    const config = {
+      whaleDefenseFundWei: "100",
+      defenseActivationPercent: 50,
+      liquidationLiveness: 1000,
+      minSponsorSize: 10
+    };
+    let strat = Strategy(config, { toBN, BN }, (...args) => events.emit("log", ...args));
+    const position = {
+      withdrawalRequestPassTimestamp: 1000,
+      numTokens: "10000",
+      sponsor: "0x1234"
+    };
+    // listen for logs
+    const eventlist = [];
+    events.on("log", (severity, data) => {
+      assert(data.message.includes("extending withdraw deadline"));
+      eventlist.push({ severity, data });
```

Comment:
here is where you can listen for events and capture them with a standard event emitter.